fitbenchmarking/core/fitting_benchmarking.py

```diff
@@ -33,661 +33,544 @@
+        :type label: str
+        """
+        self._options = options
+        self._data_dir = data_dir
+        self._checkpointer = checkpointer
+        self._label = label
+        self._results = []
+        self._failed_problems = []
+        self._unselected_minimizers = {}
+        self.__start_values_index = 0
+        self.__grabbed_output = output_grabber.OutputGrabber(self._options)
+        self.__emissions_tracker = EmissionsTracker() \
+            if 'emissions' in options.table_type else None
+
+    def benchmark(self):
+        """
+        Call benchmarking on user input and list of paths.
+        The benchmarking structure is:
+
+        .. code-block:: python
+
+            __loop_over_starting_values()
+                __loop_over_software()
+                    __loop_over_minimizers()
+                        __loop_over_jacobians()
+                            __loop_over_hessians()
+                                __perform_fit()
+
+        :return: all results,
+                 problems where all fitting failed, and
+                 minimizers that were unselected due to algorithm_type
+        :rtype: list[fibenchmarking.utils.fitbm_result.FittingResult],
+                list[str],
+                dict[str, list[str]]
+        """
+        problem_group = misc.get_problem_files(self._data_dir)
+
+        problems, results, name_count,  = [], [], {}
+
+        for p in problem_group:
+            try:
+                with self.__grabbed_output:
+                    LOGGER.info('Parsing problems')
+                    parsed = parse_problem_file(p, self._options)
+                    parsed.correct_data()
+            except FitBenchmarkException as e:
+                LOGGER.info("Could not parse problem from: %s", p)
+                LOGGER.warning(e)
+            else:
+                name_count[parsed.name] = name_count.get(parsed.name, 0) + 1
+                problems.append((p, parsed))
 
-    #################################
-    # Loops over benchmark problems #
-    #################################
-    results, failed_problems, unselected_minimizers = \
-        loop_over_benchmark_problems(problem_group,
-                                     options=options,
-                                     checkpointer=checkpointer)
+        LOGGER.info('Running problems')
 
-    checkpointer.finalise_group(label=label,
-                                failed_problems=failed_problems,
-                                unselected_minimizers=unselected_minimizers)
+        benchmark_pbar = tqdm(problems, colour='green',
+                              desc="Benchmark problems",
+                              unit="Benchmark problem", leave=True)\
+            if self._options.pbar else problems
 
-    return results, failed_problems, unselected_minimizers
+        name_index = {key: 0 for key in name_count}
 
+        LOGGER.info('Running problems')
```

Comment:
This LOGGER.info is repeated twice, here and a few lines above, so we could remove this one

```suggestion

```